test/16-fairness-tax.ts

```diff
@@ -0,0 +1,346 @@
+        const { dev, user1 } = await getNamedSigners(hre)
+        const summitToken = await getSummitToken()
+
+        await cartographerSetParam.setTokenIsNativeFarm({
+            dev,
+            tokenAddress: summitToken.address,
+            tokenIsNativeFarm: true
+        })
+
+        const taxBPEarlier = await cartographerGet.taxBP(user1.address, summitToken.address)
+        expect(taxBPEarlier).to.equal(0)
+
+        await cartographerMethod.deposit({
+            user: user1,
+            tokenAddress: summitToken.address,
+            elevation: OASIS,
+            amount: e18(5),
+        })        
+
+        const afterDepositTimestamp = await getTimestamp()
+        await mineBlockWithTimestamp(afterDepositTimestamp + days(8))
+
+        const taxBPAfter = await cartographerGet.taxBP(user1.address, summitToken.address)
+        expect(taxBPAfter).to.equal(0)        
+    })
+
+    it(`TAX BP: The fairness tax decreases correctly to the minimum value, over the duration of the tax decay`, async function() {
+        const { dev, user1 } = await getNamedSigners(hre)
+        const cakeToken = await getCakeToken()
+
+        const taxBPEarlier = await cartographerGet.taxBP(user1.address, cakeToken.address)
+        expect(taxBPEarlier).to.equal(100)
+
+        await cartographerSetParam.setTokenWithdrawTax({
+            dev,
+            tokenAddress: cakeToken.address,
+            taxBP: 500
+        })
+
+        await cartographerMethod.deposit({
+            user: user1,
+            tokenAddress: cakeToken.address,
+            elevation: OASIS,
+            amount: e18(5),
+        })
+
+        const afterDepositTimestamp = await getTimestamp()
+        await mineBlockWithTimestamp(afterDepositTimestamp + days(7))
+
+        const taxBPAfter = await cartographerGet.taxBP(user1.address, cakeToken.address)
+        expect(taxBPAfter).to.equal(100)
+    })
+
+    it(`WAIVED TAX: The fairness tax is waived when a use elevates their SUMMIT from the elevation farms to the Expedition`, async function() {
+        const { user1 } = await getNamedSigners(hre)
+        const cakeToken = await getSummitToken()
+
+        await cartographerMethod.deposit({
+            user: user1,
+            tokenAddress: cakeToken.address,
+            elevation: OASIS,
+            amount: e18(5),
+        })
+        await everestMethod.lockSummit({
+            user: user1,
+            amount: e18(1),
+            lockDuration: days(7),
+        })
+
+        const userTotalStakedAmountBefore = await cartographerGet.userTokenStakedAmount(user1.address, cakeToken.address)
+        await cartographerMethod.elevateAndLockStakedSummit({
+            user: user1,
+            elevation: OASIS,
+            amount: e18(5),
+        })
+        const userTotalStakedAmountAfter = await cartographerGet.userTokenStakedAmount(user1.address, cakeToken.address)
+
+        expect(userTotalStakedAmountAfter + 5).to.equal(userTotalStakedAmountBefore)
```

Comment:
this should be `userTotalStakedAmountAfter.add(e18(5))` equals user total staked amount before